probe/endpoint/reporter.go

```diff
@@ -14,6 +14,7 @@ import (
 
 // Node metadata keys.
 const (
+	Name = "name"
```

Comment:
Yes, if you can detangle the web of dependency and use a consistent constant like NodeMetadataName, please do so. 

But it may be (and I fear that) the "name" literal is e.g. written as if it were NodeMetadataName, but then later read as if it were SomeOtherKindOfName. And things just kind of work, but accidentally. If that's true, I think it's much better to leave it as a string literal. At least that way we know it's bad, and we're not tricking ourselves into a false sense of safety.